src/test/java/DataExtractionServiceImplTest.java

```diff
@@ -0,0 +1,31 @@
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.junit.Assert;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.List;
+import exception.DataNotAvailableException;
+import exception.InvalidFileNameException;
+import service.impl.DataExtractionServiceImpl;
+
+public class DataExtractionServiceImplTest {
+
+  @Test(expected = DataNotAvailableException.class)
+  public void testExtractDataFromSheet() throws IOException, DataNotAvailableException, InvalidFileNameException {
```

Comment:
test method should not have throws, make sure to expect exception which is expected in the given scenario of the unit test case.

also read about writing test cases, normally we have 
//given

//when

//then